### PR TITLE
Update Release note and prepare 1.9.4 release

### PR DESCRIPTION
**Pull request by yongtang**
This PR updates release notes and prepare 1.9.4 release.

This PR is part of #5603 .

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: notes/coredns-1.9.4.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+title = "CoreDNS-1.9.3 Release"
+description = "CoreDNS-1.9.4 Release Notes."
+tags = ["Release", "1.9.4", "Notes"]
+release = "1.9.4"
+date = "2022-09-07T00:00:00+00:00"
+author = "coredns"
++++
+
+This is a release with many new features. The most notable addition is a new plugin tsig for validating
+TSIG requests and signing responses. This release also adds lots of enhancements and bug fixes.
```
Comment (chrisohaver) [inline notes/coredns-1.9.4.md]:
```suggestion
TSIG requests and signing responses. This release also adds lots of enhancements and bug fixes.
```

**Comment by yongtang** [comment]
Intend to merge so that 1.9.4 can be released.

**Comment by mqasimsarfraz** [comment]
@yongtang regrading #5556 we added a [note](https://github.com/coredns/coredns/blob/master/plugin/header/README.md?plain=1#L21) related to header plugin syntax. Would it makes sense to please add a line related to that in changelog?

**Comment by yongtang** [comment]
@mqasimsarfraz There is an entry in the note entry:
```
* plugin/header: add support for query modification (https://github.com/coredns/coredns/pull/5556)
```

also updated with additional mentioning at the beginning.

**Comment by mqasimsarfraz** [comment]
> @mqasimsarfraz There is an entry in the note entry:
> 
:+1: 
> also updated with additional mentioning at the beginning.

Perfect. Thanks!

**Pull request merged**